backend/Origam.Schema.GuiModel/Designer/ControlSetItem.cs

```diff
@@ -27,359 +27,304 @@
+					throw new ArgumentOutOfRangeException(
+						"lookupId", lookupId, ResourceUtils.GetString(
+							"ErrorLookupNotFound", Name, RootItem.ItemType, 
+							RootItem.Name));
 				}
 			}
-
 			if(reportId != Guid.Empty)
 			{
 				try
 				{
-					AbstractSchemaItem item = this.PersistenceProvider.RetrieveInstance(typeof(AbstractSchemaItem), new ModelElementKey(reportId)) as AbstractSchemaItem;
+					var item = PersistenceProvider.RetrieveInstance(
+						typeof(AbstractSchemaItem), 
+						new ModelElementKey(reportId)) as AbstractSchemaItem;
 					dependencies.Add(item);
 				}
 				catch
 				{
-					throw new ArgumentOutOfRangeException("reportId", reportId, ResourceUtils.GetString("ErrorReportNotFound", this.Name, this.RootItem.ItemType, this.RootItem.Name));
+					throw new ArgumentOutOfRangeException(
+						"reportId", reportId, ResourceUtils.GetString(
+							"ErrorReportNotFound", Name, RootItem.ItemType, 
+							RootItem.Name));
 				}
 			}
-
 			if(graphicsId != Guid.Empty)
 			{
 				try
 				{
-					AbstractSchemaItem item = this.PersistenceProvider.RetrieveInstance(typeof(AbstractSchemaItem), new ModelElementKey(graphicsId)) as AbstractSchemaItem;
+					var item = PersistenceProvider.RetrieveInstance(
+						typeof(AbstractSchemaItem), 
+						new ModelElementKey(graphicsId)) as AbstractSchemaItem;
 					dependencies.Add(item);
 				}
 				catch
 				{
-					throw new ArgumentOutOfRangeException("graphicsId", graphicsId, ResourceUtils.GetString("ErrorGraphicsNotFound", this.Name, this.RootItem.ItemType, this.RootItem.Name));
+					throw new ArgumentOutOfRangeException("graphicsId", 
+						graphicsId, ResourceUtils.GetString(
+							"ErrorGraphicsNotFound", Name, RootItem.ItemType, 
+							RootItem.Name));
 				}
 			}
-
 			if(workflowId != Guid.Empty)
 			{
 				try
 				{
-					AbstractSchemaItem item = this.PersistenceProvider.RetrieveInstance(typeof(AbstractSchemaItem), new ModelElementKey(workflowId)) as AbstractSchemaItem;
+					var item = PersistenceProvider.RetrieveInstance(
+						typeof(AbstractSchemaItem), 
+						new ModelElementKey(workflowId)) as AbstractSchemaItem;
 					dependencies.Add(item);
 				}
 				catch
 				{
-					throw new ArgumentOutOfRangeException("workflowId", workflowId, ResourceUtils.GetString("ErrorWorkflowNotFound", this.Name, this.RootItem.ItemType, this.RootItem.Name));
+					throw new ArgumentOutOfRangeException("workflowId", 
+						workflowId, ResourceUtils.GetString(
+							"ErrorWorkflowNotFound", Name, RootItem.ItemType, 
+							RootItem.Name));
 				}
 			}
-
 			base.GetExtraDependencies (dependencies);
 		}
 		#endregion
 
 		#region ISchemaItemFactory Members
 
-		public override Type[] NewItemTypes
+		public override Type[] NewItemTypes => new[]
 		{
-			get
-			{
-                return new Type[] { };
-			}
-		}
-
-		public override AbstractSchemaItem NewItem(Type type, Guid schemaExtensionId, SchemaItemGroup group)
+			typeof(PropertyValueItem),
```

Comment:
Won't this cause Architect to offer new child items creation? I am not sure if this was not the idea so these options would be "hidden".

Reply:
I'm going to rollback this change.